### ISSUE TITLE
[windows][proposal] pre-check the path for spaces before building and provide an error instead of trying to build

### ISSUE THREAD

**Issue by TahaTesser** [flutter/flutter#75611]
Some users are getting `Microsoft.CppCommon.targets(234,5): error MSB6006: "cmd.exe"` when they have a space(s) in their app path.

### Propoosal
This doesn't reproduce every time but for some users, after a discussion with @stuartmorgan, It makes sense to check for spaces in the path before building windows in case "CMake or build-driving logic not escaping spaces sufficiently"

**Comment by stuartmorgan** [comment]
> after a discussion with @stuartmorgan, It makes sense to check for spaces in the path before building windows in case "CMake or build-driving logic not escaping spaces sufficiently"

What I actually said was that *if it can't be fixed* it would make sense to have an error. Ideally, it should work; it was fixed for Linux recently.

**Comment by TahaTesser** [comment]
@stuartmorgan 
That's true, a proposal in case it doesn't work, sorry, paraphrasing, updated the desc :)

**Comment by christopherfujino** [comment]
can we check for spaces and escape them? https://stackoverflow.com/questions/3328199/how-do-i-specify-a-path-with-white-space-in-it-with-cmake

**Comment by Toglefritz** [comment]
Another issue that can produce this same error message is related to the assets specified in the pubspec.yaml. If a directory in the assets list does not include a trailing slash (/), the build will result in this "MSB6006" error. For example, lets say I have a Flutter project with images stored in a directory called "assets." If I include this directory in the pubspec.yaml file like this,

  assets:
    - assets

the build will fail with the error message, "error MSB6006: "cmd.exe" exited with code 1".

![Untitled](https://user-images.githubusercontent.com/5925531/190884140-db08c864-83e3-4422-9fe3-f3138d4bb70c.png)

The fix for this is to include a trailing slash after directories in the assets list and, following fixing this missing slash, running a "flutter pub get" command.

**Comment by drumkiller-peter** [comment]
I was too having this issue and thanks to @Toglefritz , i was missing "/" at the end of the assets list in pubspec.yaml.
